### ISSUE TITLE
calc_loss_fingers nan in test_synthesis.py

### ISSUE THREAD

**Issue by braunjon** [anindita127/IMoS#9]
Hi,

When running `test_synthesis.py` the loss `calc_loss_fingers` https://github.com/anindita127/IMoS/blob/7742d061e7a0f38c729405e37449c5c90901e81c/src/test/test_synthesis.py#L451 is always nan for some test sequences. For instance for the sequence `data/post_processed/test/s10_hand_pass_1.npz`. The reason is that computing `torch.mean(torch.tensor([]))` over an empty tensor returns nan. The tensor is empty because the `r_flag` or `l_falg` is false for all entries:
https://github.com/anindita127/IMoS/blob/7742d061e7a0f38c729405e37449c5c90901e81c/src/test/test_synthesis.py#L188

Do you know how I can fix that?

I appreciate your help!

**Comment by braunjon** [comment]
Could someone confirm that they also get NaN in the calc_loss_fingers?

Thanks!

**Comment by anindita127** [comment]
Hi, we have set a hard contact threshold (https://github.com/anindita127/IMoS/blob/7742d061e7a0f38c729405e37449c5c90901e81c/src/test/test_synthesis.py#L172C8-L172C9) which does not abide with all our current sequences (but is accurate for most sequences). You can play around with that threshold and then the r_flag and l_flag will not be false for all values. I think that will solve the nan value problem. Hope this helps!

**Comment by braunjon** [comment]
Hi, could you quantify "most sequences"? For me it is 40/114 that have no nan loss on the test set s10 when using the default threshold.

**Comment by anindita127** [comment]
Hi, thanks for pointing it out. You can keep the contact threshold as 0.01 and use this condition : 
(https://github.com/anindita127/IMoS/blob/cc84feb3ad4ac221d9927a1f07a6102bb91df460/src/test/test_synthesis.py#L188C13-L189C66)

**Comment by braunjon** [comment]
Ah thanks for the update! While this avoids the NaN loss, I think the loss will be just 0 instead, leading to the same result: that the hand pose is not optimized.
